.github/workflows/preview-docs.yml

```
@@ -0,0 +1,162 @@
+name: Preview Documentation (Azure Blob Storage)
+
+on:
+    pull_request:
+        types: [opened, synchronize, reopened, closed]
+        branches:
+            - '**'
+
+permissions:
+    contents: read
+    pull-requests: write
+
+jobs:
+    build_and_deploy_job:
+        if: github.event_name == 'pull_request' && github.event.action != 'closed'
+        runs-on: ubuntu-latest
+        name: Build and Deploy Job
+        outputs:
+            doc_url: ${{ steps.deploy.outputs.docs_url }}
+        steps:
+            ## --- SETUP --- ##
+            - name: Check out code
+              uses: actions/checkout@v4
+              with:
+                  fetch-depth: 0
+
+            - name: Use Node LTS version
+              uses: actions/setup-node@v4
+              with:
+                  node-version: 20
+                  cache: yarn
+
+            - name: Enable Corepack
+              run: corepack enable
+
+            - name: Generate PR hash
+              id: pr_hash
+              run: |
+                  # Use just PR number so each commit overwrites the previous deployment
+                  pr_hash="pr-${{ github.event.pull_request.number }}"
+                  echo "hash=${pr_hash}" >> $GITHUB_OUTPUT
```

It’s best practice to wrap `$GITHUB_OUTPUT` in quotes (as per [GitHub’s own recommendation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-output-parameter)):
```suggestion
                  echo "hash=${pr_hash}" >> "$GITHUB_OUTPUT"
```

For example, if this output has espaces and is used in a shell command, it can break.